node/src/db/db_handlers.rs

```diff
@@ -0,0 +1,687 @@
+
+            let signature = match Signature::from_str(row.get::<String, _>("signature").as_str()) {
+                Ok(v) => v,
+                Err(e) => {
+                    return Err(DBErrors::TupleAtrributeParsingError {
+                        error: e.to_string(),
+                        attribute: "signature".to_string(),
+                    })
+                }
+            };
+
+            let broadcast_timestamp =
+                match bitcoin::blockdata::locktime::absolute::MedianTimePast::from_u32(
+                    row.get::<u32, _>("broadcast_timestamp"),
+                ) {
+                    Ok(val) => val,
+                    Err(error) => {
+                        return Err(DBErrors::TupleAtrributeParsingError {
+                            error: error.to_string(),
+                            attribute: "broadcastTimestamp".to_string(),
+                        })
+                    }
+                };
+
+            bead_id = row.get::<i64, _>("id");
+            fetched_bead.block_header.bits = bits;
+            fetched_bead.block_header.merkle_root = merkle;
+            fetched_bead.block_header.nonce = nonce;
+            fetched_bead.block_header.prev_blockhash = prev;
+            fetched_bead.block_header.time = time;
+            fetched_bead.block_header.version = version;
+            fetched_bead.uncommitted_metadata.signature = signature;
+            fetched_bead.uncommitted_metadata.extra_nonce = extra_nonce;
+            fetched_bead.uncommitted_metadata.broadcast_timestamp = broadcast_timestamp;
+            fetched_bead.committed_metadata.comm_pub_key = comm_pub_key;
+            fetched_bead.committed_metadata.min_target = min_target;
+            fetched_bead.committed_metadata.miner_ip = miner_ip;
+            fetched_bead.committed_metadata.payout_address = payout_address;
+            fetched_bead.committed_metadata.start_timestamp = start_timestamp;
+            fetched_bead.committed_metadata.weak_target = weak_target;
+
+            Ok(())
+        })
+        .fetch_optional(&db_connection_arc.lock().await.clone())
+        .await
+    {
+        Ok(_rows) => {
+            println!("Bead with given bead hash fetched successfully");
+        }
+        Err(error) => {
+            return Err(DBErrors::TupleNotFetched {
+                error: error.to_string(),
+            });
+        }
+    };
+    let mut bead_txs: Vec<Transaction> = Vec::new();
+    if bead_id != -1 {
+        let rows = match sqlx::query("SELECT  txid,bead_id FROM Transactions WHERE bead_id = ?")
+            .bind(bead_id)
+            .fetch_all(&db_connection_arc.lock().await.clone())
+            .await
+        {
+            Ok(rows) => rows,
+            Err(error) => {
+                return Err(DBErrors::TupleNotFetched {
+                    error: error.to_string(),
+                });
+            }
+        };
+        for tx_row in rows {
```

Comment:
- High: node/src/db/db_handlers.rs:438 – fetch_bead_by_bead_hash rebuilds the bead from the in-memory braid instead of the rows it just
    fetched. On a cold start the braid is empty, so braid_data.beads.get((bead_id - 1) as usize).unwrap() will panic (and even for non-zero
    IDs the bead_id - 1 underflows for the first bead). Even if it doesn’t crash, the function breaks after the first transaction row and
    never assigns bead_txs back to fetched_bead, so you can’t hydrate stored transactions/parents from the database.